jwst/pathloss/tests/test_pathloss.py

```diff
@@ -0,0 +1,332 @@
+"""
+Unit tests for pathloss correction
+"""
+
+from jwst.datamodels import MultiSlitModel, PathlossModel, IFUImageModel
+from jwst.pathloss import PathLossStep
+from jwst.pathloss.pathloss import (calculate_pathloss_vector,
+                                    get_aperture_from_model,
+                                    get_center, 
+                                    interpolate_onto_grid,
+                                    is_pointsource)
+from jwst.pathloss.pathloss import do_correction 
+import numpy as np
+import pytest
+
+
+def test_get_center_ifu():
+    """get_center assumes IFU targets are centered @ (0.0, 0.0)
+    """
+
+    x_pos,y_pos = get_center("NRS_IFU", None)
+
+    assert(x_pos==y_pos==0.0)
+
+
+def test_get_center_attr_err():
+    """if no center provided for modes that are not IFU,
+    center is assigned to (0.0, 0.0)
+    """
+    datmod = MultiSlitModel()
+    x_pos, y_pos = get_center("NRS_MSASPEC", datmod)
+
+    assert(x_pos==y_pos==0.0)
+
+
+def test_get_center_exp_type():
+    """if exp_type is not in NRS, center is returned (0.0,0.0) 
+    """
+    datmod = MultiSlitModel()
+    x_pos, y_pos = get_center("NRC_IMAGE", datmod)
+
+    assert(x_pos==y_pos==0.0)
+
+
+def test_get_center_exptype():
+    """ If exptype is "NRS_MSASPEC" | "NRS_FIXEDSLIT" | "NRS_BRIGHTOBJ" and
+    source_xpos and source_ypos exist in datamod.slits, make sure it's returned
+    """
+    
+    datmod = MultiSlitModel()
+    datmod.slits.append({'source_xpos':1, 'source_ypos':2})
+    
+    for exptype in ["NRS_MSASPEC", "NRS_FIXEDSLIT", "NRS_BRIGHTOBJ"]:
+        x_pos, y_pos = get_center(exptype, datmod.slits[0])
+
+        assert(x_pos==1)
+        assert(y_pos==2)
+
+
+# Begin get_aperture_from_model tests
+def test_get_app_from_model_null():
+    """If exp_type isn't the NRS or NIS specific mode,
+    routine returns None
+    """
+
+    datmod = MultiSlitModel()
+    datmod.meta.exposure.type = 'NRC_IMAGE'
+
+    result = get_aperture_from_model(datmod, None)
+
+    assert(result==None)
+
+
+def test_get_aper_from_model_fixedslit():
+    """For a given exposures aperture, make sure the correct
+    aperture reference data is returned for fixedslit mode
+    """
+
+    datmod = PathlossModel()
+    datmod.apertures.append({'name':'S200A1'})
+    datmod.meta.exposure.type = 'NRS_FIXEDSLIT'
+
+    result = get_aperture_from_model(datmod, 'S200A1')
+
+    assert(result == datmod.apertures[0])
+
+
+def test_get_aper_from_model_msa():
+    """For a given exposures aperture, make sure the correct
+    aperture reference data is returned for MSA mode
+    """
+
+    datmod = PathlossModel()
+    datmod.apertures.append({'shutters':1})
```

Comment:
It looks to me as if the comment in pathloss.py is incorrect with regard to NRS_MSASPEC.  The comment says that `match` is the number of shutters, but it's the shutter number instead.  I would suggest using a number other than 1 for this test (and most other tests, for that matter), e.g. 5.  That would check whether `result` was the shutter number or the number of shutters.

Reply:
@philhodge it looks like it is the number of shutters. `nshutters` is returned from `jwst.assign_wcs.util.get_num_msa_open_shutters` and then passed to `get_aper_from_model` in the `do_correction` portion of the script. https://github.com/spacetelescope/jwst/blob/master/jwst/pathloss/pathloss.py#L194 So, assuming that the value returned from `get_num_msa_open_shutters` matches the value of `shutters` for the slit. Since I am unit testing `get_aper_from_model` I am hardcoding the `nshutters` value to make them match on purpose. Should I unit test a case with `jwst.assign_wcs.util.get_num_msa_open_shutters` or should we leave that to the regression test for `do_correction`?

Reply:
@mfixstsci I think this is fine as it is.